helpers/loanHelpers/userCollection.tsx

```diff
@@ -440,98 +329,50 @@ async function handleFormatMarket(
   honeyClient: HoneyClient,
   honeyMarket: HoneyMarket,
   connection: Connection,
-  parsedReserves?: any
+  parsedReserves: TReserve,
+  mData?: any
 ) {
-  // calculates total market debt, total market deposits, decodes parsed reserve
-  const { totalMarketDebt, totalMarketDeposits, parsedReserve } =
-    await decodeReserve(honeyMarket, honeyClient, parsedReserves);
-  // calculates total value of a market
+  const totalMarketDebt = mData
+    ? await mData.getReserveState().outstandingDebt
+    : 0;
+
+  const totalMarketDeposits = mData
+    ? await mData.getReserveState().totalDeposits
+    : 0;
+
+  const { utilization, interestRate } =
+    await collection.marketData[0].reserves[0].getUtilizationAndInterestRate();
+
   const totalMarketValue = totalMarketDeposits + totalMarketDebt;
-  // calculates nft price of a market
-  const nftPrice = await calcNFT(
-    honeyMarket.reserves,
-    parsedReserve,
-    honeyMarket,
-    connection
-  );
-  // fetch allowance
-  const allowance = await fetchAllowance(
-    nftPrice ? nftPrice : 0,
-    1,
-    honeyUser,
-    honeyMarket.reserves
-  );
 
-  const userDebt = await fetchUserDebt(honeyUser, honeyMarket.reserves);
-  const ltv = await fetchLTV(userDebt, nftPrice ? nftPrice : 0);
-  const tvl = nftPrice ? await fetchTVL(nftPrice, obligations) : 0;
-  const userTotalDeposits = await calculateUserDeposits(
-    honeyMarket.reserves,
-    honeyUser
+  const nftPrice = await honeyMarket.fetchNFTFloorPriceInReserve(0);
+  collection.nftPrice = nftPrice;
+
+  const allowanceAndDebt = await honeyUser.fetchAllowanceAndDebt(
+    0,
+    'mainnet-beta'
   );
 
-  // if request comes from liquidation page we need the collection object to be different
-  if (origin === 'LIQUIDATIONS') {
-    collection.name;
-    collection.allowance = allowance;
-    collection.userDebt = userDebt;
-    collection.available = totalMarketDeposits;
-    collection.value = totalMarketValue;
-    collection.connection = connection;
-    collection.utilizationRate = Number(
-      f(totalMarketDebt / (totalMarketDeposits + totalMarketDebt))
-    );
-    collection.user = honeyUser;
-    collection.nftPrice = nftPrice;
-    collection.ltv = ltv;
-    collection.tvl = tvl;
+  const tvl = new BN(nftPrice * (await fetchTVL(obligations)));
```

Comment:
move BN to sdk?

Reply:
doesn't need to be BN - converted it back, simply is a calculation now - but will discuss this to ship to SDK